### ISSUE TITLE
min/max function ignoring stride of resval tensor

### ISSUE THREAD

**Issue by ahmadh84** [torch/torch7#1074]
I spotted a bug in torch.min/max functions when giving the result tensor as an argument:
`torch.max(resval, x, dim)`

It seems like the `resval` is assigned values without taking the stride into account. Here's a minimal example:
```
x = torch.rand(3,2)         -- we would like to get the row-wise minimum from this tensor
z = torch.zeros(3,4)        -- result would be stored in columns
torch.max(z[{{},1}], x, 2)  -- should store the row-wise max. in the first column of z
```
This is one of the outputs:
```
th> x
 0.5848  0.3633
 0.1953  0.5182
 0.8872  0.8088
[torch.DoubleTensor of size 3x2]

th> z
 0.5848  0.5182  0.8872  0.0000
 0.0000  0.0000  0.0000  0.0000
 0.0000  0.0000  0.0000  0.0000
[torch.DoubleTensor of size 3x4]
```
